hack/update-staging-godeps-dockerized.sh

```diff
@@ -75,8 +75,26 @@ function updateGodepManifest() {
     rm -rf Godeps # remove the current Godeps.json so we always rebuild it
     GOPATH="${TMP_GOPATH}:${GOPATH}:${GOPATH}/src/k8s.io/kubernetes/staging" godep save ${GODEP_OPTS} ./... 2>&1 | sed 's/^/  /'
 
-    # Rewriting Godeps.json to remove commits that don't really exist because we haven't pushed the prereqs yet
-    go run "${KUBE_ROOT}/staging/godeps-json-updater.go" --godeps-file="${TMP_GOPATH}/src/k8s.io/${repo}/Godeps/Godeps.json" --override-import-path="k8s.io/${repo}"
+    # Rewriting Godeps.json to cross-out commits that don't really exist because we haven't pushed the prereqs yet
+    local repo
+    for repo in $(ls -1 ${KUBE_ROOT}/staging/src/k8s.io); do
+      # remove staging prefix
+      jq '.Deps |= map(.ImportPath |= ltrimstr("k8s.io/kubernetes/staging/src/"))' Godeps/Godeps.json |
+
+        # x-out staging repo revisions. They will only be known when the publisher bot has created the final export.
+        # We keep the staging dependencies in here though to give the publisher bot a way to detect when the staging
+        # dependencies changed. If they have changed, the bot will run a complete godep restore+save. If they didn't
```

Comment:
What does "incomplete" mean? Does it mean that the robot will read the commit hash from the local repo in the GOPATH?

Reply:
if you import a package (or some repo) you have not imported before, the Godeps.json before this PR did not change (if the same other package of the same repo was already imported). We only run a complete godep restore+save if the Godeps.json changes (if not, we only replace the staging repo sha1 with a simple sed).